tools/sourcemap-tools/src/SourceProcessor.ts

```diff
@@ -236,19 +256,35 @@ export class SourceProcessor {
             ? path.resolve(path.dirname(sourceMapPath), sourceMap.sourceRoot)
             : path.resolve(path.dirname(sourceMapPath));
 
-        const sourcesContent: string[] = [];
-        for (const sourcePath of sourceMap.sources) {
+        const succeeded: string[] = [];
+        const skipped: string[] = [];
+        const failed: string[] = [];
+
+        const sourcesContent: string[] = sourceMap.sourcesContent ?? [];
```

Comment:
just checking:
if we skip the source content, because someone added a js file without type definition, will sources array be bigger than sourcesContent array?

Reply:
no, I don't push to this array, I set the indices directly. If the file does not have an original source, the respective source content will be `null`.